code/game/machinery/vending.dm

```diff
@@ -408,6 +407,19 @@
 	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
 	popup.open()
 
+GLOBAL_LIST_EMPTY(vending_machine_icon)
+
+/obj/machinery/vending/proc/GetIconForProduct(datum/data/vending_product/P)
+	var/producticon = GLOB.vending_machine_icon[P.product_name]

```

Comment:
I'm honestly surprised `GLOB.vending_machine_icon[P.product_name]` works. Would it work with the path as well?

Less possible ambiguity.